### PR TITLE
Feature deidservice first release

### PR DESCRIPTION
**Pull request by chrisribe**
First release of deidservice CLI commands, needs review.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az deidservice 

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/deidservice/azext_deidservice/tests/latest/test_deidservice.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    """Generate a random name that matches the required pattern."""
+    if len(prefix) > length:
+        raise ValueError('The length of the prefix must not be longer than random name length')
+    
+    padding_size = length - len(prefix)
+    if padding_size < 4:
+        raise ValueError('The randomized part of the name is shorter than 4, which may not be able to offer enough randomness')
+    
+    random_part = ''.join(random.choices(string.ascii_lowercase + string.digits, k=padding_size))
+    return prefix + random_part
```
Comment (kairu-ms) [inline src/deidservice/azext_deidservice/tests/latest/test_deidservice.py:26]: Please reference this doc for random name https://github.com/Azure/azure-cli/blob/dev/doc/authoring_tests.md#sample-6-random-name-and-name-mapping

Comment (chrisribe) [reply src/deidservice/azext_deidservice/tests/latest/test_deidservice.py]: Fixed, thanks!

I got lost here trying to figure out what the error was (accepted name format of the service vs random name).
The VCR.py error "AssertionError: Can't overwrite existing cassette" threw me in a loop.

The authoring_tests.md page should be re-written a bit, I may take a stab at it.

**Comment by chrisribe** [comment]
> Hi @chrisribe, Please write the description of changes which can be perceived by customers into HISTORY.rst. If you want to release a new extension version, please update the version in setup.py as well.

setup.py and HISTORY.rst seem correct, unless default version `1.0.0b1` should be changed to 0.1.0 ?

File: src/deidservice/azext_deidservice/aaz/latest/deidservice/_create.py [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+
+
+@register_command(
+    "deidservice create",
+    is_preview=True,
+)
+class Create(AAZCommand):
+    """Create a DeidService
+
+    :example: DeidServices_Create - generated by [MaximumSet] rule - generated by [MaximumSet] rule
```
Comment (kairu-ms) [inline src/deidservice/azext_deidservice/aaz/latest/deidservice/_create.py:21]: You can remove the duplicated suffix
```suggestion
    :example: DeidServices_Create
```

**Review by kairu-ms — APPROVED** [review]
The rest LGTM

File: src/deidservice/azext_deidservice/aaz/latest/deidservice/_create.py (hunk 2 above)
Comment (chrisribe) [reply src/deidservice/azext_deidservice/aaz/latest/deidservice/_create.py:21]: fixed

**Comment by chrisribe** [comment]
@kairu-ms could you take a look at the related PR here: 
https://github.com/Azure/aaz/pull/604
I cannot seem to add any reviewer to the PR for some reason...

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**